.changeset/nice-drinks-do.md

```diff
@@ -0,0 +1,9 @@
+---
+"@hashicorp/design-system-codemods": minor
+"@hashicorp/design-system-components": minor
+---
+
+`Dropdown::ListItem::Interactive`
+- Adds a yielded block.
+- Yields the `Hds::Badge` component.
+- Deprecates the `@text` argument.
```

Comment:
Question: why deprecate `@text`? Doesn't that make this component different from the same named components _not_ in the dropdown?

Reply:
Are you asking about if it differs from `Hds::Interactive`? That component doesn't accept `@text` either, so one of the motivators was to make the API the same afaik.